src/arkode/arkode_mristep.c

```diff
@@ -1865,8 +3119,35 @@ int mriStep_CheckCoupling(ARKodeMem ark_mem)
     return (ARK_INVALID_TABLE);
   }
 
+  /* Check that coupling table has compatible type */
+  if (step_mem->implicit_rhs && step_mem->explicit_rhs &&
+      (step_mem->MRIC->type != MRISTEP_IMEX) &&
+      (step_mem->MRIC->type != MRISTEP_MRISR))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid coupling table for an IMEX problem!");
+    return (ARK_ILL_INPUT);
+  }
+  if (step_mem->explicit_rhs && (step_mem->MRIC->type != MRISTEP_EXPLICIT) &&
+      (step_mem->MRIC->type != MRISTEP_IMEX) &&
+      (step_mem->MRIC->type != MRISTEP_MERK) &&
+      (step_mem->MRIC->type != MRISTEP_MRISR))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid coupling table for an explicit problem!");
+    return (ARK_ILL_INPUT);
+  }
+  if (step_mem->implicit_rhs && (step_mem->MRIC->type != MRISTEP_IMPLICIT) &&
+      (step_mem->MRIC->type != MRISTEP_IMEX) &&
+      (step_mem->MRIC->type != MRISTEP_MRISR))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid coupling table for an implicit problem!");
+    return (ARK_ILL_INPUT);
+  }
+
   /* Check that the matrices are defined appropriately */
-  if (step_mem->implicit_rhs && step_mem->explicit_rhs)
+  if (step_mem->MRIC->type == MRISTEP_IMEX)
```

Comment:
I think that there are too many "special cases" for IMEX-MRI-GARK methods, since the table structure itself determines the algorithm, whereas MRISR and MERK methods have a pre-defined structure that is independent of the table nonzero pattern.  I'll add separate [simpler/cleaner] checks for the MRISR and MERK tables instead.